addons/website_sale/controllers/main.py

```diff
@@ -1781,6 +1782,14 @@ def shop_payment_validate(self, sale_order_id=None, **post):
 
         if order and not order.amount_total and not tx_sudo:
             order.with_context(send_email=True).with_user(SUPERUSER_ID).action_confirm()
+            auto_invoice = str2bool(
+                    request.env['ir.config_parameter'].sudo().get_param('sale.automatic_invoice')
+                    )
+            if auto_invoice:
```

Comment:
I am not sure but this feels wrong :confused: 
I think what you need to do is change something in actual `auto_invoice` flow and take this into account but as you try to do it was creating payment entry and here we don't want to create payment entries :neutral_face:
Still need dive in deeper :crab: